src/chk/chk_internal.h

```diff
@@ -475,4 +493,58 @@ void chk_vos_init(void);
 
 void chk_vos_fini(void);
 
+static inline bool
+chk_rank_in_list(d_rank_list_t *rlist, d_rank_t rank)
+{
+	int	i;
+	bool	found = false;
+
+	/* XXX: if the rank list is sorted, then we can search more efficiently. */
+
+	for (i = 0; i < rlist->rl_nr; i++) {
+		if (rlist->rl_ranks[i] == rank) {
+			found = true;
+			break;
+		}
+	}
+
+	return found;
+}
+
+static inline bool
+chk_remove_rank_from_list(d_rank_list_t *rlist, d_rank_t rank)
```

Comment:
[Nit] Just FYI, there's d_rank_list_del, which may be heavy weight, but I suspect we don't care too much.

Reply:
I plan to make the ranks list for DAOS check to be sorted, then we can use binary search to locate related rank quickly.